.github/workflows/publish_docker_images.yml

```diff
@@ -9,101 +9,89 @@ on:
       - main
 
 jobs:
-  docker:
+  push_osdk_image:
     runs-on: ubuntu-latest
     steps:
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
-        with:
-          driver: docker
-
-      - name: Login to Docker Hub
-        uses: docker/login-action@v2
-        with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
-
       - uses: actions/checkout@v4
-
-      - name: Check if Docker image exists
-        id: check-images
+      - name: Setup and Check
+        id: setup-check
         run: |
-          IMAGE_TAG=$(cat DOCKER_IMAGE_VERSION)
-          if docker manifest inspect asterinas/osdk:${IMAGE_TAG} > /dev/null 2>&1; then
-              echo "osdk_image_exists=true" >> $GITHUB_ENV
-          else
-              echo "osdk_image_exists=false" >> $GITHUB_ENV
-          fi
-          if docker manifest inspect asterinas/osdk:${IMAGE_TAG}-tdx > /dev/null 2>&1; then
-              echo "osdk_tdx_image_exists=true" >> $GITHUB_ENV
-          else
-              echo "osdk_tdx_image_exists=false" >> $GITHUB_ENV
-          fi
-          if docker manifest inspect asterinas/asterinas:${IMAGE_TAG} > /dev/null 2>&1; then
-              echo "asterinas_image_exists=true" >> $GITHUB_ENV
-          else
-              echo "asterinas_image_exists=false" >> $GITHUB_ENV
-          fi
-          if docker manifest inspect asterinas/asterinas:${IMAGE_TAG}-tdx > /dev/null 2>&1; then
-              echo "asterinas_tdx_image_exists=true" >> $GITHUB_ENV
-          else
-              echo "asterinas_tdx_image_exists=false" >> $GITHUB_ENV
-          fi
-
-      - name: Fetch versions in the repo
-        id: fetch-versions
-        run: |
-          ASTER_VERSION=$(cat DOCKER_IMAGE_VERSION)
-          RUST_VERSION=$(grep -m1 -o 'nightly-[0-9]\+-[0-9]\+-[0-9]\+' rust-toolchain.toml)
-          echo "aster_version=$ASTER_VERSION" >> "$GITHUB_OUTPUT"
-          echo "rust_version=$RUST_VERSION" >> "$GITHUB_OUTPUT"
+          ./tools/github_workflows/publish_docker_images.sh ${{ secrets.DOCKERHUB_USERNAME }} ${{ secrets.DOCKERHUB_TOKEN }} osdk
```

Comment:
The script's name is misleading. From the name, it seems that the script alone completes the job of "build" and "push". But this is not true; it is more like a preparation step. I suggest renaming it `before_docker_build_and_push.sh` or `prepare_for_docker_build_and_push`.